Bug144.dfy(40,9): Error: assertion might not hold

Dafny program verifier finished with 3 verified and 1 failed assertions
